### PR TITLE
Set up Sentry in Clerks

### PR DESCRIPTION
**Pull request by JustinMi**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

Some of this description is copy-pasted from my last PR, since they are very similar: 

I integrated Sentry 🤖

In `App.js`, I added a couple of lines at the top to initiate Sentry. In `app.json`, I added some specifications for a post-publish hook. I created a `logUtils.js` file to hold all logging functions. In `ClerkLoginScreen.js`, I added a snippet to capture the current user's ID and also log if something bad happens. I also added the Sentry config info to .env.example.

### Online sources

https://forum.sentry.io/t/set-release-after-init/6759/2

## How to review

[//]: # "The order in which to review files and what to expect when testing locally"
To test, you gotta first install Sentry. I think you can just `yarn install` on my branch, but if that doesn't work, then do `yarn add sentry-expo`. 

Next, log into Sentry using the login info pinned in #chaos. Navigate to the `dccentralkitchen` project, where all the issues are logged. That's the page where you will check if Sentry actually catches errors.

Next, add Sentry config info into to your `.env.development`. You can find them in `.env.example`. They look like this:
```
...
SENTRY_ORG='...'
SENTRY_PROJECT='...'
SENTRY_AUTH_TOKEN='...'
```
You can use the `SENTRY_AUTH_TOKEN` is an org-wide token on Sentry, not user-specific. That way if we add/remove people to Sentry we don't gotta fiddle with token stuff. To find it: assuming you have access to the Blueprint org on Sentry, then go to sentry.io, and hit: `Settings` -> `Developer Settings` -> `Internal Integrations` -> `DC Central Kitchen Auth Token` -> `Tokens` to find it.

Finally, create an error. To do that, just make an incorrect login. If you check the [project page on Sentry](https://sentry.io/organizations/calblueprint/issues/?project=4525076), it should show up. 

## Next steps

There's some leftover stuff I need to still finish from the customer app's Sentry. Once I get those ironed out, I'll add them to Clerks as well. 

### Screenshots

What an Issue looks like in Sentry: 
image

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie @FranCogMents 

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: lib/logUtils.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  const { screen, action, error } = errorObject;
+  Sentry.withScope((scope) => {
+    scope.setExtra('screen', screen);
+    scope.setExtra('action', action);
+    Sentry.captureException(error);
+  });
+}
+
+export function logAuthErrorToSentry(errorObject) {
+  const { screen, action, attemptedPhone, attemptedPass, error } = errorObject;
```
[the remark this answers is not in the document]
Comment (wangannie) [reply lib/logUtils.js:13]: Change attemptedPass to attemptedPin  -- currently, it's just logging `[undefined]` bc of the mismatch.

File: screens/ClerkLoginScreen.js [hunk 2]
```diff
@@ -59,17 +62,34 @@ export default class ClerkLoginScreen extends React.Component {
...
           break;
         default:
           return;
       }
+
+      if (lookupResult.errorMsg !== '') {
+        logAuthErrorToSentry({
+          screen: 'ClerkLoginScreen',
+          action: 'handleSubmit',
+          attemptedPin: '',
```
Comment (wangannie) [inline screens/ClerkLoginScreen.js:81]: Make this `this.state.password`

File: screens/ClerkLoginScreen.js [hunk 3]
```diff
@@ -59,17 +62,34 @@ export default class ClerkLoginScreen extends React.Component {
...
-          console.log('Database malformed! Two users found');
+          errMsg = 'Database malformed! Two users found';
           break;
         default:
           return;
       }
+
+      if (lookupResult.errorMsg !== '') {
+        logAuthErrorToSentry({
+          screen: 'ClerkLoginScreen',
```
Comment (wangannie) [inline screens/ClerkLoginScreen.js:79]: Add the store name they attempted to log into. Just the PIN doesn't give us any useful info

File: screens/ClerkLoginScreen.js [hunk 4]
```diff
@@ -59,17 +62,34 @@ export default class ClerkLoginScreen extends React.Component {
...
+          action: 'handleSubmit',
+          attemptedPin: '',
+          error: errMsg,
+        });
+        console.log(errMsg);
+      } else {
+        Sentry.configureScope((scope) => {
+          scope.setUser({
```
Comment (wangannie) [inline screens/ClerkLoginScreen.js:87]: The user logging isn't working `clerkRecord` is an object. You need to pull `clerkRecord.id` and also please add `username: clerkRecord.clerkName`

File: App.js [hunk 5]
```diff
@@ -4,9 +4,18 @@ import { Asset } from 'expo-asset';
 import * as Font from 'expo-font';
 import React, { useState } from 'react';
 import { Platform, StatusBar, StyleSheet } from 'react-native';
+import * as Sentry from 'sentry-expo';
 import AppNavigator from './navigation/AppNavigator';
 import { Container } from './styled/shared';
 
+Sentry.init({
+  dsn: 'https://9b39ef5b02394775a6e939254ed55a0e@o306199.ingest.sentry.io/4525076',
+  enableInExpoDevelopment: true,
```
Comment (wangannie) [inline App.js:13]: Change this to `false` by default so we don't use up too many of BP's events, make release `v1.1.1`, and environment switch on env with `process.env.NODE_ENV`

File: .env.example [hunk 6]
```diff
@@ -1,8 +1,14 @@
...
 
 ## Needed for airtable-schema-generator if using auto mode
 AIRTABLE_EMAIL='YOUR-EMAIL-HERE'
-AIRTABLE_PASSWORD='YOUR-PASSWORD-HERE'
\ No newline at end of file
+AIRTABLE_PASSWORD='YOUR-PASSWORD-HERE'
+
+## For Sentry logging
+SENTRY_ORG='calblueprint'
+SENTRY_PROJECT='dccentralkitchen'
```
Comment (wangannie) [inline .env.example:13]: This is the wrong project. Is this info actually used anywhere? It doesn't seem like it is.

Comment (JustinMi) [reply .env.example:13]: not sure--probably not? It was something that was included in the setup tutorial

**Comment by JustinMi** [comment]
@wangannie I think there's a better way to log stores and clerks (would involve modifying `lookupClerk` in `loginUtils` but for the sake of time that can be done after the bug bash

**Pull request merged**